### ISSUE TITLE
Add Split by Depth/3P/"Problem Exp" logic to DataSplitter class

### ISSUE THREAD

**Issue by morriscb** [AllenInstitute/DeepCell#27]
Currently, the DataSplitter class used to sample the  ROIs for the training/testing the classifier splits by ROI. After our proposal to the science team was accepted, we want to change that to sampling by experiment with the requirement that it preserve the span of Depth, 3P experiments, and 2P "Problem" experiments in the sample.

After discussing with @aamster the design for the ticket comes in a few parts:

- [x] Create a `ModelInput`s similar class to read in experiment metadata containing at least experiment_id, depth, type (2P vs 3P). Metadata exists as a json file.
- [x] Modify `DataSplitter` class to take in a list of these `ExpModelInput`s and produce ROIs datasets sampled from experiment in groupings defined by the metadata. This requires modifying the methods `get_train_test_split`, `get_cross_val_split`, and `_sample_dataset`. The first two require modifying them to use the experiment labels in the `StratifiedShuffleSplit` while the latter is modified to instead take in selected experiments and return an `ROIDataset` of ROIs within those experiments.
- [x] Add input metadata json file to `TrainData` schema.

Validation
- [x] Test that expected experiment labels are created from metadata json.
- [x] Create tests for model creation from json.
- [x] Modify tests in test_data_splitter as needed.
